include/circt/Dialect/LTL/LTLOps.td

```diff
@@ -0,0 +1,312 @@
+    Preconditions the checking of the `$consequent` property on the
+    `$antecedent` sequence. In a nutshell, if the `$antecedent` sequence matches
+    at a given point in time, the `$consequent` property is checked starting at
+    the point in time at which the matched sequence ends. The result property of
+    the `ltl.implication` holds if the `$consequent` holds. Conversely, if the
+    `$antecedent` does *not* match at a given point in time, the result property
+    trivially holds. This is conceptually identical to the implication operator
+    →, but with additional temporal semantics.
+  }];
+}
+
+def EventuallyOp : LTLOp<"eventually", [Pure]> {
+  let arguments = (ins LTLAnyPropertyType:$input);
+  let results = (outs LTLPropertyType:$result);
+  let assemblyFormat = [{
+    $input attr-dict `:` type($input)
+  }];
+
+  let summary = "Ensure that a property will hold at some time in the future.";
+  let description = [{
+    Checks that the `$input` property will hold at a future time.
+  }];
+}
+
+//===----------------------------------------------------------------------===//
+// Clocking
+//===----------------------------------------------------------------------===//
+
+// Edge behavior enum for always block.  See SV Spec 9.4.2.
+
+/// AtPosEdge triggers on a rise from 0 to 1/X/Z, or X/Z to 1.
+def AtPosEdge: I32EnumAttrCase<"Pos", 0, "posedge">;
+/// AtNegEdge triggers on a drop from 1 to 0/X/Z, or X/Z to 0.
+def AtNegEdge: I32EnumAttrCase<"Neg", 1, "negedge">;
+/// AtEdge is syntactic sugar for AtPosEdge or AtNegEdge.
+def AtEdge   : I32EnumAttrCase<"Both", 2, "edge">;
+
+def ClockEdgeAttr : I32EnumAttr<"ClockEdge", "clock edge",
+                                [AtPosEdge, AtNegEdge, AtEdge]> {
+  let cppNamespace = "circt::ltl";
+}
+
+def ClockOp : LTLOp<"clock", [
+  Pure, InferTypeOpInterface, DeclareOpInterfaceMethods<InferTypeOpInterface>
+]> {
+  let arguments = (ins LTLAnyPropertyType:$input, ClockEdgeAttr:$edge, I1:$clock);
+  let results = (outs LTLSequenceOrPropertyType:$result);
+  let assemblyFormat = [{
+    $input `,` $edge $clock attr-dict `:` type($input)
+  }];
+
+  let summary = "Specify the clock for a property or sequence.";
+  let description = [{
+    Specifies the `$edge` on a given `$clock` to be the clock for an `$input`
+    property or sequence. All cycle delays in the `$input` implicitly refer to a
+    clock that advances the state to the next cycle. The `ltl.clock` operation
+    provides that clock. The clock applies to the entire property or sequence
+    expression tree below `$input`, up to any other nested `ltl.clock`
+    operations.
+
+    The operation returns a property if the `$input` is a property, and a
+    sequence otherwise.
+  }];
+}
+
+def DisableOp : LTLOp<"disable", [Pure]> {
+  let arguments = (ins LTLAnyPropertyType:$input, I1:$condition);
+  let results = (outs LTLPropertyType:$result);
+  let assemblyFormat = [{
+    $input `if` $condition attr-dict `:` type($input)
```

Comment:
I haven't thought about this carefully, but the SVA syntax chose to use the `iff` abbreviation, which is undoubtedly obscure to most people. Here, you have it spelled just "if", and I wonder if the reversed implication has any bearing on anything.

Reply:
I have no idea why SVA specifically chose `iff`. The assembly format is pretty random and won't be seen by any user or C++ programmer working with the op. I think we have other ops that take a condition which is prefixed by `if` to make it easier to read, so I just stuck with that which requires no explanation why this op specifically uses `iff` 😉

Reply:
`if` and `iff` have different truth tables classically.  sv.16.12.7 says `implies` and `iff` do in verilog too.